cosmic-usage-core/src/main/java/com/github/missioncriticalcloud/cosmic/usage/core/services/TokenService.java

```diff
@@ -0,0 +1,17 @@
+package com.github.missioncriticalcloud.cosmic.usage.core.services;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.github.missioncriticalcloud.cosmic.usage.core.model.DomainToken;
+
+public interface TokenService {
+
+    String encryptDomainToken(final String domain) throws JsonProcessingException;
+
+    String encryptDomainToken(final String domain, final long ttl) throws JsonProcessingException;
```

Comment:
I'd remove this method as the `TTL` should be specified directly in the `DomainToken`, in my opinion. Default `TTL` value could turn into a constant in `DomainToken`.